File: trab3/python/bissecao.py
from base import *


@algoritmo
def bissecao(self, f, xi, xf):
    if f(xi) * f(xf) > 0:
        raise ErroIntervalo
    for _ in xrange(self.nmi):
        xm = (xi + xf) / 2
        if abs(xf - xi) < self.tol and abs(f(xm)):
            return xm
        (xi, xf) = (xi, xm) if f(xm) * f(xi) < 0 else (xm, xf)
    raise ErroNMI

File: trab3/ruby/bissecao.rb
# encoding: utf-8
# Algoritmo BISSEÇÃO
require "./base"

Calcnum::algoritmo :bissecao do |xi, xf|
  break :erro_intervalo if f(xi) * f(xf) > 0
  iterar do
    xm = (xi + xf) / 2
    break xm if (xf - xi).abs < tol and f(xm).abs < tol
    xi, xf = f(xm) * f(xi) < 0 ? [xi, xm] : [xm, xf]
  end
end

